state/accountsDBApi.go

```diff
@@ -190,6 +190,31 @@ func (accountsDB *accountsDBApi) RecreateTrie(options common.RootHashHolder) err
 	return nil
 }
 
+// RecreateTrieIfNeeded is used to reload the trie based on the provided options only if the root hash is different than the current one
+func (accountsDB *accountsDBApi) RecreateTrieIfNeeded(options common.RootHashHolder) error {
+	accountsDB.mutRecreatedTrieBlockInfo.Lock()
+	defer accountsDB.mutRecreatedTrieBlockInfo.Unlock()
+
+	if check.IfNil(options) {
```

Comment:
this should be moved before mutex lock.. also, should we simply return nil instead of err?